client/src/extension.ts

```diff
@@ -48,6 +48,48 @@ export function activate(context: ExtensionContext) {
 		}
 	};
 
+	workspace.getConfiguration().update('yaml.format.enable', true, ConfigurationTarget.Global);
+	workspace.getConfiguration().update('yaml.trace.server', 'verbose', ConfigurationTarget.Global);
+	workspace.getConfiguration().update('yaml.validate', false, ConfigurationTarget.Global);
+	workspace.getConfiguration().update('yaml.customTags', [
+		'!And',
+		'!If',
+		'!Not',
+		'!Equals',
+		'!Or',
+		'!FindInMap',
+		'!Base64',
+		'!Cidr',
+		'!Ref',
+		'!Sub',
+		'!GetAtt',
+		'!GetAZs',
+		'!ImportValue',
+		'!Select',
+		'!Split',
+		'!Join'
+	], ConfigurationTarget.Global);
+	workspace.getConfiguration().update('[yaml]', {
+		"editor.insertSpaces": true,
+		"editor.tabSize": 2,
+		"editor.quickSuggestions": {
+			"other": true,
+			"comments": false,
+			"strings": true
+		},
+		"editor.autoIndent": true
+	}, ConfigurationTarget.Global);
+	workspace.getConfiguration().update('yaml.schemas', {
+		"https://s3.amazonaws.com/cfn-resource-specifications-us-east-1-prod/schemas/2.15.0/all-spec.json": "*.yaml"
```

Comment:
@benkehoe  do you think `*.yaml` (not `*-template.yaml`) is the correct way to do this?  I usually don't name my templates `-template.yaml`.  

I left it for the JSON portion but that is because I can't turn off validation for JSON.  To your original request I would rather keep this focused on YAML and not JSON.

Reply:
Maybe a good way to do this is `*-template.yaml` + `template.yaml`? That probably covers the bases. Could it additionally be a configuration to this extension, if people wanted to specify their own patterns?

Reply:
Lot of CloudFormation templates without `template` in the basename 

Should probably also support `.yml` and maybe `.template`

Reply:
`.template` is tricky too as some folks used to use that extension for JSON.  I'm going to test it out to make sure things don't go too crazy.

As we talk about this I'm debating a bunch of configuration options that can be adjusted if needed.

Thanks both of you for the feedback.